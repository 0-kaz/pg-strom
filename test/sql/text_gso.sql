--#
--#       Gpu Text Sort TestCases. 
--#  [TODO] Do not test commented-out queries until GPUSort supports TOAST data process. 
--#         If will support it, please remake expected outs and test_init.sql
--#

set gpu_setup_cost=0;
set random_page_cost=1000000;   --# force off index_scan.   
set enable_gpusort to on;                                                                                                                                       
set client_min_messages to warning;


select * from (select row_number() over (order by char_x desc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
select * from (select row_number() over (order by char_x asc) as rowid,id from strom_string_test) as t where t.rowid%100=0;

select * from (select row_number() over (order by nchar_x desc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
select * from (select row_number() over (order by nchar_x asc) as rowid,id from strom_string_test) as t where t.rowid%100=0;

select * from (select row_number() over (order by vchar_x desc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
select * from (select row_number() over (order by vchar_x asc) as rowid,id from strom_string_test) as t where t.rowid%100=0;

-- select * from (select row_number() over (order by nvchar_x desc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
-- select * from (select row_number() over (order by nvchar_x asc) as rowid,id from strom_string_test) as t where t.rowid%100=0;

-- select * from (select row_number() over (order by text_x desc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
-- select * from (select row_number() over (order by text_x asc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
