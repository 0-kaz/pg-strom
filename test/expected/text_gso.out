--#
--#       Gpu Text Sort TestCases. 
--#  [TODO] Do not test commented-out queries until GPUSort supports TOAST data process. 
--#         If will support it, please remake expected outs and test_init.sql
--#
set gpu_setup_cost=0;
set random_page_cost=1000000;   --# force off index_scan.   
set enable_gpusort to on;                                                                                                                                       
set client_min_messages to warning;
select * from (select row_number() over (order by char_x desc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
 rowid | id  
-------+-----
   100 | 799
   200 | 391
   300 | 325
   400 |  24
   500 | 667
   600 | 715
   700 | 419
   800 | 576
   900 | 888
  1000 | 872
(10 rows)

select * from (select row_number() over (order by char_x asc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
 rowid | id  
-------+-----
   100 | 806
   200 | 143
   300 | 400
   400 | 379
   500 | 523
   600 | 155
   700 |   1
   800 |  57
   900 | 961
  1000 | 800
(10 rows)

select * from (select row_number() over (order by nchar_x desc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
 rowid | id  
-------+-----
   100 | 643
   200 | 233
   300 | 272
   400 | 516
   500 | 549
   600 | 181
   700 | 230
   800 | 475
   900 | 713
  1000 | 449
(10 rows)

select * from (select row_number() over (order by nchar_x asc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
 rowid | id  
-------+-----
   100 | 935
   200 | 585
   300 | 236
   400 | 778
   500 | 708
   600 |  37
   700 | 570
   800 | 104
   900 | 604
  1000 | 849
(10 rows)

select * from (select row_number() over (order by vchar_x desc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
 rowid | id  
-------+-----
   100 | 386
   200 | 852
   300 | 261
   400 |  71
   500 | 352
   600 | 496
   700 | 395
   800 |  19
   900 | 411
  1000 |  59
(10 rows)

select * from (select row_number() over (order by vchar_x asc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
 rowid | id  
-------+-----
   100 | 755
   200 | 952
   300 | 398
   400 | 188
   500 | 170
   600 | 642
   700 | 700
   800 | 833
   900 | 588
  1000 | 351
(10 rows)

-- select * from (select row_number() over (order by nvchar_x desc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
-- select * from (select row_number() over (order by nvchar_x asc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
-- select * from (select row_number() over (order by text_x desc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
-- select * from (select row_number() over (order by text_x asc) as rowid,id from strom_string_test) as t where t.rowid%100=0;
